Linked list with void pointers.c

```diff
@@ -5,29 +5,24 @@ typedef struct node
 {
     void *data;
     struct node *next;
+    int (*funptr)();
 } node;
-struct node *head = NULL;
-struct node *tail = NULL;
-void insert(node **head, node **tail, const void * pointer, size_t size)
+typedef struct note
 {
-    if (*head == NULL)
-    {
-        *head = malloc(sizeof(node));
-        (*head)->data = malloc(size);
-        memcpy((*head)->data, pointer, size);
-        (*head)->next = NULL;
-        *tail = *head;
-    }
-    else
-    {
-        node *temp;
-        temp = malloc(sizeof(node));
-        temp->data = malloc(size);
-        memcpy(temp->data, pointer, size);
-        temp->next = *tail;
-        (*tail) = temp;
- 
-    }
+    struct node *head ;
+    struct node *tail ;
+} note;
+
+void insert(node **tail,void(*display_datatype)(node *tail), const void * pointer, size_t size)

```

Comment:
pass get_size as parameter not size

Reply:
void insert(node **tail,void(*display_datatype)(node *tail), const void * pointer, int (* get_size)(void * data))